mhkit/wave/contours.py

```diff
@@ -1521,4 +1522,331 @@ def _bivariate_KDE(x1, x2, bw, fit, nb_steps, Ndata_bivariate_KDE, kwargs):
+
+    return_periods = np.asarray(return_periods)
+    contour_probs = 1 / (365*24*60*60/sea_state_duration * return_periods)
+
+    # Reliability contour generation
+    # Calculate reliability
+    beta_lines = stats.norm.ppf((1 - contour_probs), 0, 1)
+    # Add zero as lower bound to first contour
+    beta_lines = np.hstack((0, beta_lines))
+    # Discretize the circle
+    theta_lines = np.linspace(0, 2 * np.pi, 1000)
+    # Add probablity of 1 to the reliability set, corresponding to
+    # probability of the center point of the normal space
+    contour_probs = np.hstack((1, contour_probs))
+
+    # Vary U1,U2 along circle sqrt(U1^2+U2^2) = beta
+    u1_lines = np.dot(np.cos(theta_lines[:, None]), beta_lines[None, :])
+
+    # Removing values on the H_s = 0 line that are far from the circles in the
+    # normal space that will be evaluated to speed up calculations
+    minval = np.amin(u1_lines) - 0.5
+    mask = c1_normzeroline > minval
+    c1_normzeroline = c1_normzeroline[mask]
+    c2_normzeroline = c2_normzeroline[mask]
+
+    # Transform to polar coordinates
+    theta_zeroline = np.arctan2(c2_normzeroline, c1_normzeroline)
+    rho_zeroline = np.sqrt(c1_normzeroline**2 + c2_normzeroline**2)
+    theta_zeroline[theta_zeroline < 0] = theta_zeroline[
+        theta_zeroline < 0] + 2 * np.pi
+
+    sample_alpha, sample_beta, weight_points = _generate_sample_data(
+        beta_lines, rho_zeroline, theta_zeroline, points_per_interval,
+        contour_probs)
+
+    # Sample transformation to principal component space
+    sample_u1 = sample_beta * np.cos(sample_alpha)
+    sample_u2 = sample_beta * np.sin(sample_alpha)
+
+
+    comp1_sample = stats.invgauss.ppf(
+        stats.norm.cdf(sample_u1, loc=0, scale=1),
+        mu=comp1['mu'], loc=0, scale=comp1['scale'])
+    mu_sample = mu_slope * comp1_sample + mu_intercept
+
+    # Calculate sigma values at each point on the circle
+    sigma_sample = np.polyval(sigma_polynomial_coeffcients, comp1_sample)
+
+    # Use calculated mu and sigma values to calculate C2 along the contour
+    comp2_sample = stats.norm.ppf(stats.norm.cdf(sample_u2, loc=0, scale=1),
+                                  loc=mu_sample, scale=sigma_sample)
+
+    # Sample transformation into Hs-T space
+    h_sample, t_sample = _princomp_inv(
+        comp1_sample, comp2_sample, coeff, shift)
+
+    return h_sample, t_sample, weight_points
+
+
+def samples_contour(t_samples, t_contour, hs_contour):
+    """Get Hs points along a specified environmental contour using
+    user-defined T values.
+
+    Parameters
+    ----------
+    t_samples : nparray
+        Points for sampling along return contour
+    t_contour : np.array
+        T values along contour
+    hs_contour : np.array
+        Hs values along contour
+
+    Returns
+    -------
+    hs_samples : nparray
+        points sampled along return contour
+    """
+    #finds minimum and maximum energy period values
```

Comment:
Add checks on data types and values passed.

Reply:
Done w/ @hivanov-nrel latest commit.